### PR TITLE
Inital support for RK3288

### PR DESCRIPTION
**Pull request by wom-bat**
Using the Yeacreate NTablet as the platform, this patch series adds
support for the Rockchip RK3288.

Signed-off-by: Peter Chubb <peter.chubb@data61.csiro.au>

Test with: seL4/seL4#280

### CODE REVIEW

**Comment by walle-whale** [comment]
LGTM, could you please update style.

**Review by walle-whale — APPROVED** [review]

**Comment by LukeMondy** [comment]
@ssrg-bamboo `test`
This is also a test for the new deployment of Bamboo

**Comment by ssrg-bamboo** [comment]
Hello, I'm a bot! I'll bring this PR into Trustworthy Systems and run some tests

**Comment by ssrg-bamboo** [comment]
## Test results for commit 71034360afd2b85de7646fde89bd567212f7b2bb

### Summary

| Test name | Result |
| --- | --- |
| Code Quality - Check licenses - seL4_tools - EXT-seL4-rk3288-50 | SUCCESSFUL |
| Code Quality - Reuse License Check - seL4_tools - EXT-seL4-rk3288-50 | SUCCESSFUL |
| Code Quality - Signed-off-by - seL4_tools - EXT-seL4-rk3288-50 | SUCCESSFUL |
| Code Quality - Style - seL4_tools - EXT-seL4-rk3288-50 | FAILED |
| Testing project -  lmondy - Style - seL4_tools - EXT-seL4-rk3288-50 | FAILED |

### Detailed failure logs

| Project | Stage | Result summary |
| --- | --- | --- |
| Code Quality - Style - seL4_tools - EXT-seL4-rk3288-50 | Style check on changed files | 1 of 3 jobs failed |

**Job: gitlint**
Build logs

    27-Nov-2020 21:34:30	Build Code Quality - Style - seL4_tools - EXT-seL4-rk3288-50 - gitlint #1 (CQ-SEL4TOOLSSTYLE170-SEL4TOOLSGITLINTPR-1) started building on agent A local agent 1, bamboo version: 7.1.4
    27-Nov-2020 21:34:30	I run on the Bamboo server!
    27-Nov-2020 21:34:30	Build working directory is /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-SEL4TOOLSSTYLE170-SEL4TOOLSGITLINTPR
    27-Nov-2020 21:34:30	Executing build Code Quality - Style - seL4_tools - EXT-seL4-rk3288-50 - gitlint #1 (CQ-SEL4TOOLSSTYLE170-SEL4TOOLSGITLINTPR-1)
    27-Nov-2020 21:34:30	Preparing artifact 'code' for use at /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-SEL4TOOLSSTYLE170-SEL4TOOLSGITLINTPR/. (location: .)
    27-Nov-2020 21:34:30	Artifact 'code' prepared in 24.98 ms at /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-SEL4TOOLSSTYLE170-SEL4TOOLSGITLINTPR/./code.tar.gz
    27-Nov-2020 21:34:30	Running pre-build action: VCS Version Collector
    27-Nov-2020 21:34:30	Starting task 'Untar code' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    27-Nov-2020 21:34:30	+ tar xf code.tar.gz --recursive-unlink
    27-Nov-2020 21:34:30	Finished task 'Untar code' with result: Success
    27-Nov-2020 21:34:30	Starting task 'Set permissions so Bamboo can cleanup' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    27-Nov-2020 21:34:30	+ chmod -R o+rwx /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-SEL4TOOLSSTYLE170-SEL4TOOLSGITLINTPR
    27-Nov-2020 21:34:30	Finished task 'Set permissions so Bamboo can cleanup' with result: Success
    27-Nov-2020 21:34:30	Starting task 'Lint commit messages' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    27-Nov-2020 21:34:30	+ cd seL4_tools
    27-Nov-2020 21:34:30	+ '[' -f .gitlint ']'
    27-Nov-2020 21:34:30	+ CONFIG=/var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-SEL4TOOLSSTYLE170-SEL4TOOLSGITLINTPR/standard_seL4_tools/misc/.gitlint
    27-Nov-2020 21:34:30	+ gitlint --commits 021452dd83cd85a943b56be83f71f5dc2e74d2fb..71034360afd2b85de7646fde89bd567212f7b2bb --config /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-SEL4TOOLSSTYLE170-SEL4TOOLSGITLINTPR/standard_seL4_tools/misc/.gitlint
    27-Nov-2020 21:34:30	Traceback (most recent call last):
    27-Nov-2020 21:34:30	  File "/usr/local/bin/gitlint", line 6, in <module>
    27-Nov-2020 21:34:30	    from gitlint.cli import cli
    27-Nov-2020 21:34:30	ModuleNotFoundError: No module named 'gitlint'
    27-Nov-2020 21:34:30	Failing task since return code of [/var/atlassian/application-data/bamboo/temp/CQ-SEL4TOOLSSTYLE170-SEL4TOOLSGITLINTPR-1-ScriptBuildTask-1681270970901999430.sh] was 1 while expected 0
    27-Nov-2020 21:34:30	Finished task 'Lint commit messages' with result: Failed
    27-Nov-2020 21:34:30	Starting task 'Set permissions so Bamboo can cleanup' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    27-Nov-2020 21:34:30	+ chmod -R o+rwx /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-SEL4TOOLSSTYLE170-SEL4TOOLSGITLINTPR
    27-Nov-2020 21:34:30	Finished task 'Set permissions so Bamboo can cleanup' with result: Success
    27-Nov-2020 21:34:30	Running post build plugin 'NCover Results Collector'
    27-Nov-2020 21:34:30	Running post build plugin 'Artifact Copier'
    27-Nov-2020 21:34:30	Running post build plugin 'npm Cache Cleanup'
    27-Nov-2020 21:34:30	Running post build plugin 'Clover Results Collector'
    27-Nov-2020 21:34:30	Running post build plugin 'Docker Container Cleanup'
    27-Nov-2020 21:34:30	Successfully removed working directory at '/var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/CQ-SEL4TOOLSSTYLE170-SEL4TOOLSGITLINTPR'
    27-Nov-2020 21:34:30	Finalising the build...
    27-Nov-2020 21:34:30	Stopping timer.
    27-Nov-2020 21:34:30	Build CQ-SEL4TOOLSSTYLE170-SEL4TOOLSGITLINTPR-1 completed.
    27-Nov-2020 21:34:30	Running on server: post build plugin 'NCover Results Collector'
    27-Nov-2020 21:34:30	Running on server: post build plugin 'Build Hanging Detection Configuration'
    27-Nov-2020 21:34:30	Running on server: post build plugin 'Clover Delta Calculator'
    27-Nov-2020 21:34:30	Running on server: post build plugin 'Maven Dependencies Postprocessor'
    27-Nov-2020 21:34:30	All post build plugins have finished
    27-Nov-2020 21:34:30	Generating build results summary...
    27-Nov-2020 21:34:30	Saving build results to disk...
    27-Nov-2020 21:34:30	Store variable context...
    27-Nov-2020 21:34:30	Indexing build results...
    27-Nov-2020 21:34:30	Finished building CQ-SEL4TOOLSSTYLE170-SEL4TOOLSGITLINTPR-1.


| Project | Stage | Result summary |
| --- | --- | --- |
| Testing project -  lmondy - Style - seL4_tools - EXT-seL4-rk3288-50 | Style check on changed files | 1 of 3 jobs failed |

**Job: gitlint**
Build logs

    27-Nov-2020 21:34:31	Build Testing project -  lmondy - Style - seL4_tools - EXT-seL4-rk3288-50 - gitlint #1 (LMONDY-SEL4TOOLSSTYLE4-SEL4TOOLSGITLINTPR-1) started building on agent A local agent 1, bamboo version: 7.1.4
    27-Nov-2020 21:34:31	I run on the Bamboo server!
    27-Nov-2020 21:34:31	Build working directory is /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/LMONDY-SEL4TOOLSSTYLE4-SEL4TOOLSGITLINTPR
    27-Nov-2020 21:34:31	Executing build Testing project -  lmondy - Style - seL4_tools - EXT-seL4-rk3288-50 - gitlint #1 (LMONDY-SEL4TOOLSSTYLE4-SEL4TOOLSGITLINTPR-1)
    27-Nov-2020 21:34:31	Preparing artifact 'code' for use at /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/LMONDY-SEL4TOOLSSTYLE4-SEL4TOOLSGITLINTPR/. (location: .)
    27-Nov-2020 21:34:31	Artifact 'code' prepared in 12.10 ms at /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/LMONDY-SEL4TOOLSSTYLE4-SEL4TOOLSGITLINTPR/./code.tar.gz
    27-Nov-2020 21:34:31	Running pre-build action: VCS Version Collector
    27-Nov-2020 21:34:31	Starting task 'Untar code' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    27-Nov-2020 21:34:31	+ tar xf code.tar.gz --recursive-unlink
    27-Nov-2020 21:34:31	Finished task 'Untar code' with result: Success
    27-Nov-2020 21:34:31	Starting task 'Set permissions so Bamboo can cleanup' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    27-Nov-2020 21:34:31	+ chmod -R o+rwx /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/LMONDY-SEL4TOOLSSTYLE4-SEL4TOOLSGITLINTPR
    27-Nov-2020 21:34:31	Finished task 'Set permissions so Bamboo can cleanup' with result: Success
    27-Nov-2020 21:34:31	Starting task 'Lint commit messages' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    27-Nov-2020 21:34:31	+ cd seL4_tools
    27-Nov-2020 21:34:31	+ '[' -f .gitlint ']'
    27-Nov-2020 21:34:31	+ CONFIG=/var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/LMONDY-SEL4TOOLSSTYLE4-SEL4TOOLSGITLINTPR/standard_seL4_tools/misc/.gitlint
    27-Nov-2020 21:34:31	+ gitlint --commits 021452dd83cd85a943b56be83f71f5dc2e74d2fb..71034360afd2b85de7646fde89bd567212f7b2bb --config /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/LMONDY-SEL4TOOLSSTYLE4-SEL4TOOLSGITLINTPR/standard_seL4_tools/misc/.gitlint
    27-Nov-2020 21:34:31	Traceback (most recent call last):
    27-Nov-2020 21:34:31	  File "/usr/local/bin/gitlint", line 6, in <module>
    27-Nov-2020 21:34:31	    from gitlint.cli import cli
    27-Nov-2020 21:34:31	ModuleNotFoundError: No module named 'gitlint'
    27-Nov-2020 21:34:31	Failing task since return code of [/var/atlassian/application-data/bamboo/temp/LMONDY-SEL4TOOLSSTYLE4-SEL4TOOLSGITLINTPR-1-ScriptBuildTask-7607056127708209362.sh] was 1 while expected 0
    27-Nov-2020 21:34:31	Finished task 'Lint commit messages' with result: Failed
    27-Nov-2020 21:34:31	Starting task 'Set permissions so Bamboo can cleanup' of type 'com.atlassian.bamboo.plugins.scripttask:task.builder.script'
    27-Nov-2020 21:34:31	+ chmod -R o+rwx /var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/LMONDY-SEL4TOOLSSTYLE4-SEL4TOOLSGITLINTPR
    27-Nov-2020 21:34:31	Finished task 'Set permissions so Bamboo can cleanup' with result: Success
    27-Nov-2020 21:34:31	Running post build plugin 'NCover Results Collector'
    27-Nov-2020 21:34:31	Running post build plugin 'Artifact Copier'
    27-Nov-2020 21:34:31	Running post build plugin 'npm Cache Cleanup'
    27-Nov-2020 21:34:31	Running post build plugin 'Clover Results Collector'
    27-Nov-2020 21:34:31	Running post build plugin 'Docker Container Cleanup'
    27-Nov-2020 21:34:31	Successfully removed working directory at '/var/atlassian/application-data/bamboo/xml-data/build-dir/140083221/LMONDY-SEL4TOOLSSTYLE4-SEL4TOOLSGITLINTPR'
    27-Nov-2020 21:34:31	Finalising the build...
    27-Nov-2020 21:34:31	Stopping timer.
    27-Nov-2020 21:34:31	Build LMONDY-SEL4TOOLSSTYLE4-SEL4TOOLSGITLINTPR-1 completed.
    27-Nov-2020 21:34:31	Running on server: post build plugin 'NCover Results Collector'
    27-Nov-2020 21:34:31	Running on server: post build plugin 'Build Hanging Detection Configuration'
    27-Nov-2020 21:34:31	Running on server: post build plugin 'Clover Delta Calculator'
    27-Nov-2020 21:34:31	Running on server: post build plugin 'Maven Dependencies Postprocessor'
    27-Nov-2020 21:34:31	All post build plugins have finished
    27-Nov-2020 21:34:31	Generating build results summary...
    27-Nov-2020 21:34:31	Saving build results to disk...
    27-Nov-2020 21:34:31	Store variable context...
    27-Nov-2020 21:34:31	Indexing build results...
    27-Nov-2020 21:34:31	Finished building LMONDY-SEL4TOOLSSTYLE4-SEL4TOOLSGITLINTPR-1.


This is the most I can report on right now, sorry!

**Comment by ssrg-bamboo** [comment]
All the tests we ran have passed! Nice job!

File: elfloader-tool/src/arch-arm/armv/armv7-a/32/mmu-hyp.S [hunk 1]
```diff
@@ -81,5 +81,7 @@ BEGIN_FUNC(arm_enable_hyp_mmu)
     mcr     p15, 0, r0, c1, c0, 1
 #endif /* CONFIG_MAX_NUM_NODES > 1 */
 
+	isb sy
```
Comment (yyshen) [inline elfloader-tool/src/arch-arm/armv/armv7-a/32/mmu-hyp.S:84]: you need an ISB after modifying system control registers to ensure the following instructions observe the changes to system control registers.

Comment (oliver-wm) [reply elfloader-tool/src/arch-arm/armv/armv7-a/32/mmu-hyp.S:84]: Will update :+1:

**Comment by axel-h** [comment]
What's the situation around RK3288 today? Is this still a platform to add? Maybe if somebody stands up as official maintainer?

**Comment by kent-mcleod** [comment]
Closing due to not currently having a maintainer for the new platform.

**Pull request closed without merging**